kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/istio/IstioGatewayHandler.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.kubernetes.handlers.istio;
+
+import org.ballerinax.kubernetes.exceptions.KubernetesPluginException;
+import org.ballerinax.kubernetes.handlers.AbstractArtifactHandler;
+import org.ballerinax.kubernetes.models.istio.IstioGatewayModel;
+import org.ballerinax.kubernetes.models.istio.IstioServerModel;
+import org.ballerinax.kubernetes.utils.KubernetesUtils;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinax.kubernetes.KubernetesConstants.ISTIO_GATEWAY_FILE_POSTFIX;
+import static org.ballerinax.kubernetes.KubernetesConstants.YAML;
+
+/**
+ * Generates istio gateway artifacts.
+ */
```

Comment:
Since this is a newly added file shall we add the version?

Reply:
Added.